crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -587,6 +642,27 @@ impl Store {
         Ok(cache)
     }
 
+    pub(crate) async fn transaction(&self) -> Result<StoreTransaction> {
+        StoreTransaction::new(self.clone()).await
+    }
+
+    // Note: bnjbvr lost against borrowck here. Ideally, the `F` parameter would take a
+    // `&StoreTransaction`, but callers didn't quite like that.
+    #[cfg(test)]
+    pub(crate) async fn with_transaction<
```

Comment:
Hmm, that's more what I had in mind, why is this for tests only? Can't we mimic the `sqlx` interface here? They seem to have figured out how to make the borrow checker happy: https://docs.rs/sqlx/latest/sqlx/prelude/trait.Connection.html#method.transaction.

Reply:
We can definitely use it outside tests too; I found it nice to use in tests because it creates a small scope, but in general code where the transaction may live longer, it was good enough to have an explicit commit. Both are fine to me, I don't have a strong opinion here.